Source/WebCore/css/parser/CSSPropertyParserHelpers.cpp

```diff
@@ -4420,30 +4452,61 @@ static RefPtr<CSSValue> consumeGeneratedImage(CSSParserTokenRange& range, const
     return result;
 }
 
-static RefPtr<CSSValue> consumeImageSet(CSSParserTokenRange& range, const CSSParserContext& context, OptionSet<AllowedImageType> allowedImageTypes)
+// https://w3c.github.io/csswg-drafts/css-images-4/#image-set-notation
+static RefPtr<CSSImageSetOptionValue> consumeImageSetOption(CSSParserTokenRange& args, const CSSParserContext& context, OptionSet<AllowedImageType> allowedImageTypes)
 {
-    auto rangeCopy = range;
-    auto args = consumeFunction(rangeCopy);
-    CSSValueListBuilder imageSet;
-    do {
-        auto image = consumeImage(args, context, allowedImageTypes);
-        if (!image)
-            return nullptr;
+    auto image = consumeImage(args, context, allowedImageTypes);
+    if (!image)
+        return nullptr;
+
+    auto result = CSSImageSetOptionValue::create(image.releaseNonNull());
 
-        auto option = CSSImageSetOptionValue::create(image.releaseNonNull());
+    RefPtr<CSSPrimitiveValue> resolution;
+    RefPtr<CSSPrimitiveValue> type;
 
-        auto resolution = consumeResolution(args);
-        if (resolution) {
-            if (resolution->floatValue() <= 0)
+    // Optional resolution and type in any order.
+    for (size_t i = 0; i < 2 && !args.atEnd(); ++i) {
+        if (auto optionalArg = consumeMetaConsumer<ImageSetResolutionOrTypeConsumer>(args, { }, { }, { }, { }, { })) {
```

Comment:
nit: no abbreviation, can you name this `optionalArgument` or `parameter`?

Reply:
I expanded this to `optionalArgument` and I changed the argument to the function from `args` to `range` to match most of the other consumer functions calling that parameter `range`.